### PR TITLE
Added logger and fixed test cases

### PR DESCRIPTION
**Pull request by sujata-m**
- Added logger 
 - Removed unused code
 - Refactor 
 - Fixed tests
 - Only Zip files are accepted

### CODE REVIEW

**Comment by susrisha** [comment]
Any task for this or just for additional logging?

**Review by susrisha — APPROVED** [review]

**Comment by sujata-m** [comment]
> Added Ticket https://dev.azure.com/TDEI-UW/TDEI/_workitems/edit/265

File: src/gtfs_flex_validation.py [hunk 1]
```diff
@@ -101,20 +98,21 @@ def download_single_file(self, file_upload_path=None) -> str:
                 file_path = file.file_path.split('/')[-1]
```
Comment (sureshgaussian) [inline src/gtfs_flex_validation.py:98]: change this to os.path.basename

File: src/gtfs_flex_validation.py [hunk 2]
```diff
@@ -38,16 +45,17 @@ def is_gtfs_flex_valid(self) -> tuple[Union[bool, Any], Union[str, Any]]:
         dest = DOWNLOAD_FILE_PATH
         is_valid = False
         validation_message = ''
-        if self.file_relative_path.endswith('.json') or self.file_relative_path.endswith('.zip'):
+        root, ext = os.path.splitext(self.file_relative_path)
+        if ext:
```
Comment (sureshgaussian) [inline src/gtfs_flex_validation.py:49]: why this ext check? and i dont understand the else part of this check. you should just check if the file ends with .zip and if not reject it.

**Review by sureshgaussian — CHANGES_REQUESTED** [review]

File: src/gtfs_flex_validation.py (hunk 1 above)
Comment (sujata-m) [reply src/gtfs_flex_validation.py:98]: sure

File: src/gtfs_flex_validation.py [hunk 3]
```diff
@@ -35,58 +42,25 @@ def validate(self) -> tuple[bool, str]:
...
         validation_message = ''
-        if self.file_relative_path.endswith('.json') or self.file_relative_path.endswith('.zip'):
+        root, ext = os.path.splitext(self.file_relative_path)
+        if ext and ext.lower() == '.zip':
             downloaded_file_path = self.download_single_file(self.file_path)
             try:
+                logger.info(f' Downloaded file path: {downloaded_file_path}')
                 gcv_test_release.test_release(DATA_TYPE, SCHEMA_VERSION, downloaded_file_path)
                 is_valid = True
+                GTFSFlexValidation.clean_up(downloaded_file_path)
```
Comment (sureshgaussian) [inline src/gtfs_flex_validation.py:54]: shouldn't this be after try block? in the case where the file is not valid, we will not delete the downloaded file on the current server, which will lead to explosion of local storage.

File: src/gtfs_flex_validation.py [hunk 4]
```diff
@@ -98,23 +72,24 @@ def download_single_file(self, file_upload_path=None) -> str:
...
                 return f'{DOWNLOAD_FILE_PATH}/{file_path}'
             else:
-                print('File not found!')
+                logger.info(' File not found!')
         except Exception as e:
-            print(e)
+            traceback.print_exc()
+            logger.error(e)
 
     def download_file(self, source, dest):
```
Comment (sureshgaussian) [inline src/gtfs_flex_validation.py:86]: do we need this function?

**Review by sureshgaussian — CHANGES_REQUESTED** [review]
please see the two comments from me. fix at least the first one.

File: src/gtfs_flex_validation.py (hunk 3 above)
Comment (sujata-m) [reply src/gtfs_flex_validation.py:54]: you are correct

**Pull request merged**